### PR TITLE
RS Clone – Angular Final Project

### PR DESCRIPTION
**Pull request by dzichonka**
### Project Overview :duck:

CV Generator is a modern web application built with Angular 21 and Firebase that allows users to create and manage professional CVs.
Users can fill in profile blocks (experience, education, skills, etc.) and dynamically generate multiple CV versions by selecting specific blocks.
The application uses a reactive architecture with Angular Signals and RxJS for efficient state management and real-time updates

[Task](https://github.com/rolling-scopes-school/tasks/tree/master/angular/modules/rsclone)

[Presentation](https://gamma.app/docs/CV-Generator-Final-Team-Presentation-v9keusjklgde7hy?mode=present#card-jr7gzoy6os6y74v)

### Deployment :dodo:

[Vercel deployment](https://angular-final-project-rs.vercel.app/)

- Hosting (Firebase):   script ```"deploy": "npm run build && firebase deploy"``` for custom deployment
- CI/CD: Automated builds triggered on pushes to develop.

### Project structure :goose:

Screenshot 2026-03-01 at 10 25 16

### Architecture Diagram

```
┌─────────────────────┐
│      Browser        │
│  Angular SPA (v21)  │
└──────────┬──────────┘
           │
           │ Firebase SDK
           ▼
┌─────────────────────┐
│   Firebase Auth     │
│   (JWT, session)    │
└─────────────────────┘
           │
           ▼
┌─────────────────────┐
│    Firestore DB     │
│                     │
│  profile collection │
│  cvs collection     │
└─────────────────────┘
           │
           ▼
┌─────────────────────┐
│   Firebase Hosting  │
└─────────────────────┘
```
- Angular 21 (standalone, signals)
- Firebase Authentication
- Firestore (Profile collection → CVs collection)
- Hosting (Firebase)

#### Pages
Screenshot 2026-03-01 at 15 07 59
Screenshot 2026-03-01 at 15 08 07
Screenshot 2026-03-01 at 15 08 15
Screenshot 2026-03-01 at 15 08 26
Screenshot 2026-03-01 at 15 08 35
Screenshot 2026-03-01 at 15 14 55

#### Skeleton loader
Screenshot 2026-03-01 at 15 37 12

#### i18n
- polish
Screenshot 2026-03-01 at 15 40 51

- english
Screenshot 2026-03-01 at 15 40 46

#### theme
- light
Screenshot 2026-03-01 at 15 40 05

- dark
Screenshot 2026-03-01 at 15 40 10

#### notification
- error
Screenshot 2026-02-14 at 10 57 55

- success
Screenshot 2026-02-14 at 10 58 48

#### Lazyloading with fallback
Screenshot 2026-01-29 at 23 17 16

#### Drag-and-drop

Screenshot 2026-03-01 at 15 48 29

#### Save PDF
[CV creator.pdf](https://github.com/user-attachments/files/25661976/CV.creator.pdf)

#### Backend with Firebase (Firestore)

Screenshot 2026-02-15 at 11 00 57

### Self-assessment table

feature | points | evidence |  
-- | -- | -- | --
Single source of truth with signals for one major feature | 20 | [code screenshot](https://prnt.sc/XRQaRnTe2Fdn) | 20
3+ computed values that are used in templates or logic for that feature (no duplicates) | 15 | [code screenshot](https://prnt.sc/J0pVG-tN3kJT) [code screenshot](https://prnt.sc/XRQaRnTe2Fdn) [code screenshot](https://prnt.sc/GJkxbvUVB9zX) | 15
2+ effect with clean-up (unsubscribe/teardown) that trigger a real side effect (navigation, storage, analytics, DOM/API). Console.log-only doesn’t count. | 15 | [code screenshot](https://prnt.sc/nlQVNMqT4NhW) [code screenshot](https://prnt.sc/X5pI7Cg7yR20) [code screenshot](https://prnt.sc/hvN3YSlYPPwv) | 15
Bridge RxJS ↔ Signals: convert 3+ Observables with toSignal (HTTP/realtime) and use the resulting signals in template/computed. |   |   | 15
Signal inputs (input()) in 3+ components to simplify component APIs (instead of @Input()), used in the main flow. | 10 | [code screenshot](https://prnt.sc/7Bt6-ug7R07a) [code screenshot](https://prnt.sc/4KN_pXyResYM) [code screenshot](https://prnt.sc/h5LxpAchlARH) | 10
Signal queries (viewChild, contentChild) in 1+ component where it simplifies logic (focus/measure/portal) |   |   | 5
Use untracked() or a custom equality in 1+ place to avoid extra updates; add a short code comment explaining why. |   |   | 10
No reactive loops / leaks: effects don’t re-trigger themselves; clean-up is visible in code/tests. |   |   | 10
Useful computed: each computed helps (filtering, totals, flags) and is used in ≥2 places or replaces heavy template logic. | 10 | [code screenshot](https://prnt.sc/J0pVG-tN3kJT) [code screenshot](https://prnt.sc/XRQaRnTe2Fdn) [code screenshot](https://prnt.sc/GJkxbvUVB9zX) | 10
Clear boundaries: signals for app/component state; RxJS stays where it fits (HTTP/streams). No duplicate state (Signal + Subject) without a reason. | 10 | [code screenshot](https://prnt.sc/8WETN7yVxAh6) [code screenshot](https://prnt.sc/hwMFOxQI0gAv) [code screenshot](https://prnt.sc/R5qGgt9hF5iN) | 10
Performance awareness: use track with @for or trackBy with ngFor; avoid unnecessary recomputes (note, profiler screenshot, or counter proof). | 10 | [code screenshot](https://prnt.sc/ypiJf8NIpVi7) [code screenshot](https://prnt.sc/QYL1MRKl32pR) [code screenshot](https://prnt.sc/o4pBmH6hC1GW) | 10
Tested behavior: unit test covers a computed/effect or a signals-based component; E2E covers state updates of the main journey. |   |   | 10
Short rationale: in README (5–7 lines) explain where you chose signals vs RxJS and why (clarity, perf, simplicity). | 10 | [code screenshot](https://prnt.sc/ZTKslKHsQ8Qm) | 10
Functional routes with lazy loadComponent | 25 | [code screenshot](https://prnt.sc/W0IKFrc-0IbD) | 25
Guards/resolvers with typed data | 20 | [code screenshot](https://prnt.sc/9qyGSSW7c5WZ) [code screenshot](https://prnt.sc/o_gczrRFSVGn) | 20
withComponentInputBinding() to map params to inputs | 15 | [code screenshot](https://prnt.sc/JH7Yx0-KYMeS) [code screenshot](https://prnt.sc/oeZ-ndc17wEt) [code screenshot](https://prnt.sc/kPBoj_TSxaGb) | 15
Data prefetch or custom preloading strategy |   |   | 20
Error route and 404 page, safe redirects | 10 | [code screenshot](https://prnt.sc/mrZpgZNhkBzL) | 10
Deep linking: keep query params, restore scroll, good loading UX |   |   | 20
Unit tests for key components/services/pipes with edge cases | 50 | [code screenshot](https://prnt.sc/LPgWpE5nC8TF) | 50
E2E tests for main flows (happy path + one error path) | 50 | [code screenshot](https://prnt.sc/wQT7UeMGwtO8) | 50
Mock HTTP, test interceptors and error states | 20 | [code screenshot](https://prnt.sc/OMO4mJgVbJMS) | 20
Use a component testing library/harness where it helps | 10 | [code screenshot](https://prnt.sc/PICmY41k4T7L) | 10
strict: true without ignored errors in prod code | 20 | [code screenshot](https://prnt.sc/WmhGnSAs_hQb) | 20
Good domain models with generics and type guards | 15 | [code screenshot](https://prnt.sc/tn895tRwUUne) [code screenshot](https://prnt.sc/B_TTQZzZycvb) [code screenshot](https://prnt.sc/doXnC8xV68WY) | 15
satisfies and utility types (Pick, Partial, Omit) where helpful | 5 | [code screenshot](https://prnt.sc/ujLh7Hu4JTjR) [code screenshot](https://prnt.sc/Jc6zjMcEPDA4) | 5
Feature-sliced structure (core/shared/ui/features) |   |   | 30
Reusable components with clear inputs/outputs, content projection | 20 | [code screenshot](https://prnt.sc/3vrGH-theSbb) | 20
Useful attribute or structural directives | 20 | [code screenshot](https://prnt.sc/T0u55jL-LGPj) [code screenshot](https://prnt.sc/4mw0rfXSPsUn) [code screenshot](https://prnt.sc/2SUAmS-n-PxM) | 20
DI patterns: injection tokens for config, clean boundarie |   |   | 10
Well-designed pure pipes with strong typing | 10 | [code screenshot](https://prnt.sc/4ObltcW8edbG) | 10
Typed HttpClient layer, interceptors for auth/headers |   |   | 25
Consistent error handling with user-friendly messages and safe retry policy | 20 | [code screenshot](https://prnt.sc/EIBzOt_VA3Mb) | 20
Cancel in-flight requests on navigation, avoid races |   |   | 20
Local cache with invalidation |   |   | 15
Complex form with custom validators (sync + async), clear error UX | 40 | [code screenshot](https://prnt.sc/3fR2d5p0gSiu) [code screenshot](https://prnt.sc/h-2us5A3X7bf) | 40
Dynamic fields with FormArray |   |   | 15
Save draft and restore form state | 15 | [code screenshot](https://prnt.sc/X-nuwg8FCimp) | 15
Keyboard access, labels, aria-describedby for errors | 10 | [code screenshot](https://prnt.sc/h-2us5A3X7bf) | 10
Design tokens with CSS custom properties, theme switch | 25 | [code screenshot](https://prnt.sc/e7JLouG1JEtn) | 25
Responsive layout (BreakpointObserver or modern CSS) |   |   | 15
Angular animations that improve UX |   |   | 10
Good empty/loading/error states, skeletons | 20 | [code screenshot](https://prnt.sc/9OBo-asbSX8Y) | 20
Content projection with selectors, meaningful slots |   |   | 20
ngTemplateOutlet, ng-container, new control flow to simplify DOM | 20 | [code screenshot](https://prnt.sc/Aukb9_Nod1fV) | 20
Code-splitting and lazy loading for heavy parts | 20 | [code screenshot](https://prnt.sc/a5DhIdL-K3Jh) | 20
Image lazy loading and virtual scroll for large lists |   |   | 20
Performance budget in README + measured Lighthouse gains | 20 | [code screenshot](https://prnt.sc/-ISsyk5CPLXn) [code screenshot](https://prnt.sc/4CrP3TgzBOe2) | 20
Your own backend (NestJS/Express) or Firebase (Firestore/Auth/Storage) with docs | 40 | [firebase screenshot](https://prnt.sc/isVfGRfHE2EA) | 40
Auth with JWT/OAuth2, protected routes | 20 | [firebase screenshot](https://prnt.sc/8rbtyfmWfK-8) | 20
Realtime features (WebSocket or Firestore realtime) |   |   | 20
Keyboard navigation and focus management across main screens |   |   | 20
Proper semantics and ARIA for forms, dialogs, menus | 20 | [code screenshot](https://prnt.sc/Z4-8O5pwh4ur) [code screenshot](https://prnt.sc/9h_tQmoPLXke) | 20
CI pipeline: lint + unit tests + build + preview | 20 | [code screenshot](https://prnt.sc/VxnnMZ-Wd-Qh) [code screenshot](https://prnt.sc/iJtJJTjVo6lq) | 20
Clear README with run steps, env variables, architecture diagram | 20 | [code screenshot](https://prnt.sc/TC4ZlvYix3wN) | 20
Release notes/changelog and issue templates | 10 | [code screenshot](https://prnt.sc/Tq1-XdaJ_WTG) | 10
Error monitoring (for example, Sentry) or remote logging |   |   | 10
Two languages using Angular i18n or ngx-translate | 20 | [UI screenshot](https://prnt.sc/c65A8tE885HB) | 20
Total  points | 730 |   |  

feature 	points 	evidence	
Single source of truth with signals for one major feature	20	[code screenshot](https://prnt.sc/XRQaRnTe2Fdn)	20
3+ computed values that are used in templates or logic for that feature (no duplicates)	15	[code screenshot](https://prnt.sc/J0pVG-tN3kJT) [code screenshot](https://prnt.sc/XRQaRnTe2Fdn) [code screenshot](https://prnt.sc/GJkxbvUVB9zX)	15
2+ effect with clean-up (unsubscribe/teardown) that trigger a real side effect (navigation, storage, analytics, DOM/API). Console.log-only doesn’t count.	15	"[code screenshot](https://prnt.sc/nlQVNMqT4NhW) [code screenshot](https://prnt.sc/X5pI7Cg7yR20) [code screenshot](https://prnt.sc/hvN3YSlYPPwv)
"	15
Bridge RxJS ↔ Signals: convert 3+ Observables with toSignal (HTTP/realtime) and use the resulting signals in template/computed.			15
Signal inputs (input()) in 3+ components to simplify component APIs (instead of @Input()), used in the main flow.	10	"[code screenshot](https://prnt.sc/7Bt6-ug7R07a) [code screenshot](https://prnt.sc/4KN_pXyResYM) [code screenshot](https://prnt.sc/h5LxpAchlARH)
"	10
Signal queries (viewChild, contentChild) in 1+ component where it simplifies logic (focus/measure/portal)			5
Use untracked() or a custom equality in 1+ place to avoid extra updates; add a short code comment explaining why.			10
No reactive loops / leaks: effects don’t re-trigger themselves; clean-up is visible in code/tests.			10
Useful computed: each computed helps (filtering, totals, flags) and is used in ≥2 places or replaces heavy template logic.	10	[code screenshot](https://prnt.sc/J0pVG-tN3kJT) [code screenshot](https://prnt.sc/XRQaRnTe2Fdn) [code screenshot](https://prnt.sc/GJkxbvUVB9zX)	10
Clear boundaries: signals for app/component state; RxJS stays where it fits (HTTP/streams). No duplicate state (Signal + Subject) without a reason.	10	[code screenshot](https://prnt.sc/8WETN7yVxAh6) [code screenshot](https://prnt.sc/hwMFOxQI0gAv) [code screenshot](https://prnt.sc/R5qGgt9hF5iN)	10
Performance awareness: use track with @for or trackBy with ngFor; avoid unnecessary recomputes (note, profiler screenshot, or counter proof).	10	[code screenshot](https://prnt.sc/ypiJf8NIpVi7) [code screenshot](https://prnt.sc/QYL1MRKl32pR) [code screenshot](https://prnt.sc/o4pBmH6hC1GW)	10
Tested behavior: unit test covers a computed/effect or a signals-based component; E2E covers state updates of the main journey.			10
Short rationale: in README (5–7 lines) explain where you chose signals vs RxJS and why (clarity, perf, simplicity).	10	[code screenshot](https://prnt.sc/ZTKslKHsQ8Qm)	10
Functional routes with lazy loadComponent	25	[code screenshot](https://prnt.sc/W0IKFrc-0IbD)	25
Guards/resolvers with typed data	20	[code screenshot](https://prnt.sc/9qyGSSW7c5WZ) [code screenshot](https://prnt.sc/o_gczrRFSVGn)	20
withComponentInputBinding() to map params to inputs	15	[code screenshot](https://prnt.sc/JH7Yx0-KYMeS) [code screenshot](https://prnt.sc/oeZ-ndc17wEt) [code screenshot](https://prnt.sc/kPBoj_TSxaGb)	15
Data prefetch or custom preloading strategy			20
Error route and 404 page, safe redirects	10	[code screenshot](https://prnt.sc/mrZpgZNhkBzL)	10
Deep linking: keep query params, restore scroll, good loading UX			20
Unit tests for key components/services/pipes with edge cases	50	[[code screenshot](https://prnt.sc/LPgWpE5nC8TF)](https://prnt.sc/LPgWpE5nC8TF)	50
E2E tests for main flows (happy path + one error path)	50	[code screenshot](https://prnt.sc/wQT7UeMGwtO8)	50
Mock HTTP, test interceptors and error states	20	[code screenshot](https://prnt.sc/OMO4mJgVbJMS)	20
Use a component testing library/harness where it helps	10	[code screenshot](https://prnt.sc/PICmY41k4T7L)	10
strict: true without ignored errors in prod code	20	[code screenshot](https://prnt.sc/WmhGnSAs_hQb)	20
Good domain models with generics and type guards	15	[code screenshot](https://prnt.sc/tn895tRwUUne) [code screenshot](https://prnt.sc/B_TTQZzZycvb) [code screenshot](https://prnt.sc/doXnC8xV68WY)	15
satisfies and utility types (Pick, Partial, Omit) where helpful	5	[code screenshot](https://prnt.sc/ujLh7Hu4JTjR) [code screenshot](https://prnt.sc/Jc6zjMcEPDA4)	5
Feature-sliced structure (core/shared/ui/features)			30
Reusable components with clear inputs/outputs, content projection	20	[code screenshot](https://prnt.sc/3vrGH-theSbb) 	20
Useful attribute or structural directives	20	[code screenshot](https://prnt.sc/T0u55jL-LGPj) [code screenshot](https://prnt.sc/4mw0rfXSPsUn) [code screenshot](https://prnt.sc/2SUAmS-n-PxM) 	20
DI patterns: injection tokens for config, clean boundarie			10
Well-designed pure pipes with strong typing	10	[code screenshot](https://prnt.sc/4ObltcW8edbG)	10
Typed HttpClient layer, interceptors for auth/headers			25
Consistent error handling with user-friendly messages and safe retry policy	20	[code screenshot](https://prnt.sc/EIBzOt_VA3Mb)	20
Cancel in-flight requests on navigation, avoid races			20
Local cache with invalidation 			15
Complex form with custom validators (sync + async), clear error UX	40	[code screenshot](https://prnt.sc/3fR2d5p0gSiu) [code screenshot](https://prnt.sc/h-2us5A3X7bf) 	40
Dynamic fields with FormArray			15
Save draft and restore form state	15	[code screenshot](https://prnt.sc/X-nuwg8FCimp)	15
Keyboard access, labels, aria-describedby for errors	10	[code screenshot](https://prnt.sc/h-2us5A3X7bf) 	10
Design tokens with CSS custom properties, theme switch	25	[code screenshot](https://prnt.sc/e7JLouG1JEtn) 	25
Responsive layout (BreakpointObserver or modern CSS)			15
Angular animations that improve UX			10
Good empty/loading/error states, skeletons	20	[code screenshot](https://prnt.sc/9OBo-asbSX8Y) 	20
Content projection with selectors, meaningful slots			20
ngTemplateOutlet, ng-container, new control flow to simplify DOM	20	[code screenshot](https://prnt.sc/Aukb9_Nod1fV) 	20
Code-splitting and lazy loading for heavy parts	20	[code screenshot](https://prnt.sc/a5DhIdL-K3Jh) 	20
Image lazy loading and virtual scroll for large lists			20
Performance budget in README + measured Lighthouse gains	20	[code screenshot](https://prnt.sc/-ISsyk5CPLXn) [code screenshot](https://prnt.sc/4CrP3TgzBOe2)  	20
Your own backend (NestJS/Express) or Firebase (Firestore/Auth/Storage) with docs	40	[firebase screenshot](https://prnt.sc/isVfGRfHE2EA)  	40
Auth with JWT/OAuth2, protected routes	20	[firebase screenshot](https://prnt.sc/8rbtyfmWfK-8)  	20
Realtime features (WebSocket or Firestore realtime)			20
Keyboard navigation and focus management across main screens			20
Proper semantics and ARIA for forms, dialogs, menus	20	[code screenshot](https://prnt.sc/Z4-8O5pwh4ur) [code screenshot](https://prnt.sc/9h_tQmoPLXke)  	20
CI pipeline: lint + unit tests + build + preview	20	[code screenshot](https://prnt.sc/VxnnMZ-Wd-Qh) [code screenshot](https://prnt.sc/iJtJJTjVo6lq)  	20
Clear README with run steps, env variables, architecture diagram	20	[code screenshot](https://prnt.sc/TC4ZlvYix3wN)	20
Release notes/changelog and issue templates	10	[code screenshot](https://prnt.sc/Tq1-XdaJ_WTG)	10
Error monitoring (for example, Sentry) or remote logging			10
Two languages using Angular i18n or ngx-translate	20	[UI screenshot](https://prnt.sc/c65A8tE885HB)	20
Total  points 	730

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <title>CV Generator</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/svg+xml" href="cv.svg" />
+    <meta name="description" content="An application for generating CVs." />
+    <meta property="og:title" content="CV Generator" />
+    <meta property="og:description" content="A tool for creating professional CVs." />
+    <meta property="og:type" content="website" />
+    <meta property="og:url" content="https://https://angular-final-project-rs.vercel.app//" />
+    <meta property="og:image" content="cv.svg" />
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/index.html:14]: Done

File: src/app/components/shared/header/header.component.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<header class="content">
+  <a routerLink="auth" matButton="filled"><mat-icon>login</mat-icon>Login</a>
+  <a routerLink="/" matButton="filled"><mat-icon>home</mat-icon>Main</a>
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/shared/header/header.component.html]: Done

File: src/app/services/error.service.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GlobalErrorService implements ErrorHandler {
+  private dialog = inject(MatDialog);
+
+  public handleError(error: unknown): void {
+    console.error(error);
+  }
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/services/error.service.ts]: deleted

File: src/app/components/creator-page/profile-tabs/photo-form/photo-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+export class PhotoFormComponent {
+  private dialogRef = inject(MatDialogRef);
+  private changeDetectorRef = inject(ChangeDetectorRef);
+  // private notificationService = inject(NotificationService);
+
+  public item = input<Photo | null>(null);
+  public isUploading = false;
+  public selectedFile: File | null = null;
+  public imagePreview: string | null = null;
+  constructor() {
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/creator-page/profile-tabs/photo-form/photo-form.component.ts:34]: deleted isUploading and selectedFile

File: src/app/components/creator-page/profile-block/profile-block.component.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  experience: ExperienceFormComponent,
+  about: AboutFormComponent,
+  links: LinkFormComponent,
+  languages: LanguageFormComponent,
+};
+
+@Component({
+  selector: 'app-profile-block',
+  imports: [ProfileBlockItemComponent, MatIconButton, AsyncPipe, MatIcon, AsyncPipe, MatButton],
+  templateUrl: './profile-block.component.html',
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/creator-page/profile-block/profile-block.component.ts:52]: Done priviosly

File: src/app/components/auth-page/forgot-password-page/forgot-password-page.component.html [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<main class="content">
+  <h1>Welcome to our App.</h1>
+  <p>Please enter you Email/Username to reset the password.</p>
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/auth-page/forgot-password-page/forgot-password-page.component.html]: Done priviosly

File: src/app/components/shared/dialog/create-item/create-item.component.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    if (this.form.invalid) return;
+    this.loading.set(true);
+    const form = this.form.getRawValue();
+    this.cvService.createCv(form.title).subscribe({
+      next: () => {
+        this.loading.set(false);
+        this.dialogRef.close({ saved: true });
+      },
+    });
+  }
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/shared/dialog/create-item/create-item.component.ts:53]: added
```
      error: () => {
        this.loading.set(false);
        this.errorMessages.set('An error occurred while creating the item.');
      },
```

File: src/app/components/welcome-page/welcome-page.component.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<div class="text-center mt-5 pt-5">
+  <h1>Welcome to our App.</h1>
+  <p>Please sign-in or sign-up to start using our App.</p>
+  <button matButton="tonal" class="me-2" [routerLink]="['auth/sign-in']">Sign-In</button>
+  <button matButton="filled" [routerLink]="['/auth/sign-up']">Sign-Up</button>
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/welcome-page/welcome-page.component.html:5]: added

File: src/app/components/shared/dialog/create-item/create-item.component.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  public updateErrorMessage(): void {
+    if (this.form.controls.title.hasError('required')) {
+      this.errorMessages.set('You must enter a title.');
+    } else if (this.form.controls.title.hasError('maxlength')) {
+      this.errorMessages.set('Your id must be less than 20 characters.');
+    } else if (this.form.controls.title.hasError('minlength')) {
+      this.errorMessages.set('Your id must be more than 3 characters.');
+    } else {
+      this.errorMessages.set('');
+    }
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/shared/dialog/create-item/create-item.component.ts:36]: fixed

File: src/app/components/auth-page/forgot-password-page/forgot-password-page.component.ts [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+})
+export class ForgotPasswordPageComponent {
+  private fromBuilder = inject(FormBuilder);
+  private authService = inject(AuthService);
+
+  public readonly isPasswordResetEmailSent = this.authService.isPasswordResetEmailSent;
+  public readonly errorMessage = this.authService.errorMessage;
+  public readonly isSubmissionInProgress = this.authService.isSubmissionInProgress;
+
+  public form = this.fromBuilder.nonNullable.group({
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/auth-page/forgot-password-page/forgot-password-page.component.ts]: fixed

File: src/app/components/creator-page/profile-block/profile-block.component.html [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+      </button>
+      <app-profile-block-item [itemData]="item" [blockType]="blockType()"></app-profile-block-item>
+      <button matIconButton aria-label="Delete item" type="button" (click)="removeItem(item.id)">
+        <mat-icon>delete</mat-icon>
+      </button>
+    </div>
+  }
+} @else {
+  <h2>There is no items to show</h2>
+  <p>Please add new item</p>
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/creator-page/profile-block/profile-block.component.html]: fixed

File: src/app/services/theme.service.ts [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+      localStorage.setItem(this.storageKey, 'dark');
+    }
+  }
+
+  public setTheme(theme: Theme): void {
+    localStorage.setItem(this.storageKey, theme);
+    document.body.classList.remove('light', 'dark');
+    document.body.classList.add(theme);
+    localStorage.setItem(this.storageKey, theme);
+  }
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/services/theme.service.ts:36]: deleted

File: src/app/components/creator-page/profile-tabs/experience-form/experience-form.component.html [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    <input matInput formControlName="endDate" />
+  </mat-form-field>
+
+  <mat-form-field appearance="outline">
+    <mat-label>Description</mat-label>
+    <textarea matInput rows="3" formControlName="description"></textarea>
+    @if (form.controls.description.invalid && form.controls.description.touched) {
+      <mat-error> Last name is required </mat-error>
+    }
+  </mat-form-field>
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/components/creator-page/profile-tabs/experience-form/experience-form.component.html:42]: fixed previously

File: src/index.html [hunk 14]
```diff
@@ -0,0 +1,24 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <title>CV Generator</title>
+    <base href="/" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/svg+xml" href="cv.svg" />
+    <meta name="description" content="An application for generating CVs." />
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/index.html:9]: it's ok

File: src/app/app.routes.ts [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+        canActivate: [AuthGuard],
+        data: {
+          authGuardPipe: redirectToLogin,
+        },
+      },
+      {
+        path: '',
+        component: WelcomePageComponent,
+      },
+      {
```
[the remark this answers is not in the document]
Comment (dzichonka) [reply src/app/app.routes.ts:31]: It's for unregistered users

**Review by pavlovspace — APPROVED** [review]
